src/query/plan/cost_estimator.hpp

```diff
@@ -225,20 +280,42 @@ class CostEstimator : public HierarchicalLogicalOperatorVisitor {
     return false;
   }
 
+  bool PostVisit(Produce &op) override {
+    auto scope = Scope();
+
+    // translate all the stats to the scope outside the return
+    for (const auto &symbol : op.ModifiedSymbols(table_)) {
+      auto stats = GetStatsFor(symbol);
+      if (stats.has_value()) {
+        scope.symbol_stats[symbol.name()] =
+            SymbolStatistics{.cardinality = stats.value().cardinality, .degree = stats.value().degree};
+      }
+    }
+
+    scopes_.push_back(scope);
+    return true;
+  }
+
   bool PreVisit(Apply &op) override {
```

Comment:
Apply is the subquery operator. I first calculate the cost in the main branch, and then I multiply the cardinality of the nodes from the subquery part, because the result cost is cartesian